feat: document cryptography & backup data loss prevention (#5)
toolsforhumanity/docs: world-app/backup/structure-and-sync.mdx

- **`store_file(designator, filePath)`** - Adds a new file to the backup. Verifies path uniqueness, computes checksum, triggers pack & upload.
- **`list_files(designator)`** - Returns all entries for the given designator.
- **`replace_all_files_for_designator(designator, newFilePath)`** - Replaces all existing entries for a designator with a new file.
- **`remove_file(filePath)`** - Removes a file entry from the manifest.

### Pack & Upload Process

When building and uploading a new backup version:

1. **Validate local manifest** - Error if missing; prompt to download remote.
2. **Build unsealed backup** - Collect all files from the manifest. Verify checksums match for every file.
3. **Submit sealed backup** - Include updated manifest inside. Report current and new manifest hashes to backend.
4. **Backend validation** - Verifies current hash matches recorded state before accepting update.

### Download & Unpack Process

When downloading and unpacking a remote backup:

1. **Decrypt sealed backup** - Extract and replace local manifest unconditionally.
2. **Unpack files** - Write each file to its declared `filePath`. Replace existing files (log errors for checksum mismatches).

> This approach is safe because clients enforce explicit state transitions starting from the current manifest.

Reviewer: ```suggestion
This approach is safe because clients enforce explicit state transitions based on the current manifest.
```
